### PR TITLE
Drew/profile faves

### PR DESCRIPTION
**Pull request by doctor3w**
Recommendations on the user are finally done and work with Joe's need backend API requests.

### CODE REVIEW

File: Podcast/ProfileHeaderView.swift [hunk 1]
```diff
@@ -67,7 +67,9 @@ class ProfileHeaderView: UIView {
         buttonBar = UIView(frame: CGRect(x: 0, y: profileAreaHeight, width: frame.width, height: buttonBarHeight))
         buttonBar.backgroundColor = .podcastWhite
         
-        profileImage = UIImageView(frame: .zero)
+        profileImage = ImageView(frame: .zero)
+        profileImage.frame = CGRect(x: (frame.width - profileImageWidth) / 2, y: profileImageY, width: profileImageWidth, height: profileImageWidth)
```
Comment (natashaarmbrust) [inline Podcast/ProfileHeaderView.swift]: Set the frame to be the line below on initilization (no cgrect.zero)

File: Podcast/FetchUserRecommendationsEndpointRequest.swift [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import UIKit
+import SwiftyJSON
+
+class FetchUserRecommendationsEndpointRequest: EndpointRequest {
+    var userID: String
+    
+    init(userID: String) {
+        
+        self.userID = userID
```
Comment (natashaarmbrust) [inline Podcast/FetchUserRecommendationsEndpointRequest.swift]: Delete the excess white space here

File: Podcast/ExternalProfileViewController.swift [hunk 3]
```diff
@@ -323,18 +349,23 @@ class ExternalProfileViewController: ViewController, UITableViewDataSource, UITa
...
-//        guard let favoriteEpisodes = favorites else { return 0 }
-//        return favoriteEpisodes.count
+        guard let favoriteEpisodes = favorites else { return 0 }
+        return favoriteEpisodes.count
     }
     
     func recommendedEpisodesOuterTableViewCell(cell: RecommendedEpisodesOuterTableViewCell, didSelectItemAt indexPath: IndexPath) {
-        print("Selected episode at \(indexPath.row)")
+        let episode = favorites[indexPath.row]
+        let episodeViewController = EpisodeDetailViewController()
```
Comment (natashaarmbrust) [inline Podcast/ExternalProfileViewController.swift]: Rename to episodeDetailViewCobtroller

Comment (natashaarmbrust) [reply Podcast/ExternalProfileViewController.swift]: Controller*

File: Podcast/ExternalProfileViewController.swift [hunk 4]
```diff
@@ -131,15 +135,27 @@ class ExternalProfileViewController: ViewController, UITableViewDataSource, UITa
...
+                self.favorites = results
+                
+                // Need guard in case view hasn't been created
+                guard let profileTableView = self.profileTableView else { return }
+                profileTableView.reloadData()
+            }
+        }
+        favoritesRequest.failure = { (endpointRequest: EndpointRequest) in
+            DispatchQueue.main.async {
+                // Display error
```
Comment (natashaarmbrust) [inline Podcast/ExternalProfileViewController.swift]: Delete failure block .. we will make costum alert view next semester

File: Podcast/ExternalProfileViewController.swift [hunk 5]
```diff
@@ -152,6 +168,15 @@ class ExternalProfileViewController: ViewController, UITableViewDataSource, UITa
                 profileTableView.reloadData()
             }
         }
+        subscriptionsRequest.failure = { (endpointRequest: EndpointRequest) in
+            DispatchQueue.main.async {
```
Comment (natashaarmbrust) [inline Podcast/ExternalProfileViewController.swift]: Delete failure block

File: Podcast/ExternalProfileViewController.swift [hunk 6]
```diff
@@ -131,15 +135,27 @@ class ExternalProfileViewController: ViewController, UITableViewDataSource, UITa
...
-//                guard let results = favoritesEndpointRequest.processedResponseValue as? [Episode] else { return }
-//                self.favorites = results
-//                self.profileTableView.reloadData()
-//            }
-//        }
-//        System.endpointRequestQueue.addOperation(favoritesRequest) // UNCOMMENT WHEN FAVORITES ARE DONE
+        let favoritesRequest = FetchUserRecommendationsEndpointRequest(userID: id)
+        favoritesRequest.success = { (favoritesEndpointRequest: EndpointRequest) in
+            DispatchQueue.main.async {
+                guard let results = favoritesEndpointRequest.processedResponseValue as? [Episode] else { return }
```
Comment (natashaarmbrust) [inline Podcast/ExternalProfileViewController.swift]: Why are you doing this on another thread? The request has already succeeded .. can you explain your reasoning behind this

File: Podcast/FetchUserRecommendationsEndpointRequest.swift (hunk 2 above)
Comment (doctor3w) [reply Podcast/FetchUserRecommendationsEndpointRequest.swift]: It looks nice though

File: Podcast/ExternalProfileViewController.swift (hunk 4 above)
Comment (doctor3w) [reply Podcast/ExternalProfileViewController.swift]: Can't we leave this alert and just replace it when we have a custom alert

**Pull request merged**